programs/perpetuals/src/state/pool.rs

```diff
@@ -1570,6 +1658,83 @@ mod test {
             )
             .unwrap()
         );
+
+        // Test Optimal fees
+        custody.fees.mode = FeesMode::Optimal;
+        custody.fees.fee_max = 250; // 0.025
+        custody.fees.fee_optimal = 10; // 0.001
+        assert_eq!(
+            18_000_000, /* 0.1% fee because we approach target ratio. */
+            pool.get_fee(
+                0,
+                custody.fees.add_liquidity,
+                scale(18, custody.decimals),
+                0,
+                &custody,
+                &token_price,
+            )
+            .unwrap()
+        );
+        assert_eq!(
+            4_014_000_000, /* 2.23% fee because we exceed target ratio, nearing max ratio. */
```

Comment:
may seem a little high, but in line with the spec:

<img width="1305" alt="Screenshot 2023-10-12 at 10 23 22 AM" src="https://github.com/solana-labs/perpetuals/assets/1387955/168a0dfe-7d35-425f-83a7-d235ce4cc834">